File: gems/decomplex/examples/lua/fat-union.lua
function handle(node)
  if node == AST.Call then node.line(); node.col(); node.ty(); node.span(); node.parent(); node.recv() end
  if node == AST.Func then node.line(); node.col(); node.ty(); node.span(); node.parent(); node.name() end
  if node == AST.Lit then node.line(); node.col(); node.ty(); node.span(); node.parent(); node.value() end
end

File: gems/decomplex/examples/ruby/fat-union.rb
# frozen_string_literal: true

def handle(node)
  case node
  when AST::Call
    node.line
    node.col
    node.ty
    node.span
    node.parent
    node.recv
  when AST::Func
    node.line
    node.col
    node.ty
    node.span
    node.parent
    node.name
  when AST::Lit
    node.line
    node.col
    node.ty
    node.span
    node.parent
    node.value
  end
end
